### PR TITLE
feat: compat with rxjs `7.x` and `8.x-alpha`

### LINKED ISSUE #3427: 🤖 - Deprecated rxjs symbols
### Description

image
image
image
image
image

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #3427 #3426

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: projects/cdk/tokens/active-element.ts [hunk 1]
```diff
@@ -37,6 +36,11 @@ export const TUI_ACTIVE_ELEMENT = new InjectionToken<Observable<EventTarget | nu
                 focusout$.pipe(
                     // eslint-disable-next-line rxjs/no-unsafe-takeuntil
                     takeUntil(mousedown$),
+                    /**
+                     * replace to
```
Comment (nsbarsukov) [inline projects/cdk/tokens/active-element.ts:40]: Add `TODO`

File: projects/core/components/alert/alert.component.ts [hunk 2]
```diff
@@ -71,6 +71,14 @@ export class TuiAlertComponent<O, I> implements OnInit {
         )
             .pipe(
                 takeUntil(fromEvent(this.elementRef.nativeElement, 'mouseenter')),
+                /**
+                 * replace to
```
Comment (nsbarsukov) [inline projects/core/components/alert/alert.component.ts:75]: Add `TODO`

File: projects/core/components/link/link.component.ts [hunk 3]
```diff
@@ -67,8 +67,10 @@ export class TuiLinkComponent implements TuiFocusableElementAccessor {
     focusVisible = false;
 
     readonly focusedChange = merge(
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'focusin').pipe(mapTo(true)),
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'focusout').pipe(mapTo(false)),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'focusin').pipe(map(() => true)),
```
Comment (nsbarsukov) [inline projects/core/components/link/link.component.ts:70]:
```suggestion
        tuiTypedFromEvent(this.elementRef.nativeElement, 'focusin').pipe(map(ALWAYS_TRUE_HANDLER),
```

https://github.com/Tinkoff/taiga-ui/blob/4b799b5f5fb98eeda678b5aa7af82f9901544667/projects/cdk/constants/always-true-handler.ts#L4

File: projects/core/components/link/link.component.ts [hunk 4]
```diff
@@ -67,8 +67,10 @@ export class TuiLinkComponent implements TuiFocusableElementAccessor {
...
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'focusout').pipe(
+            map(() => false),
```
Comment (nsbarsukov) [inline projects/core/components/link/link.component.ts:72]:
```suggestion
            map(ALWAYS_FALSE_HANDLER),
```

https://github.com/Tinkoff/taiga-ui/blob/4b799b5f5fb98eeda678b5aa7af82f9901544667/projects/cdk/constants/always-false-handler.ts#L4

File: projects/demo/src/modules/components/slider/examples/6/index.ts [hunk 5]
```diff
@@ -20,7 +20,7 @@ export class TuiSliderExample6 {
     readonly active$ = new BehaviorSubject(false);
     readonly showHint$ = this.active$.pipe(
         distinctUntilChanged(),
-        switchMap(active => (active ? of(true) : timer(1000).pipe(mapTo(false)))),
+        switchMap(active => (active ? of(true) : timer(1000).pipe(map(() => false)))),
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/slider/examples/6/index.ts:23]:
```suggestion
        switchMap(active => (active ? of(true) : timer(1000).pipe(map(ALWAYS_FALSE_HANDLER)))),
```

File: projects/demo/src/modules/components/svg/examples/1/index.ts [hunk 6]
```diff
@@ -16,7 +16,7 @@ import {mapTo} from 'rxjs/operators';
     encapsulation,
 })
 export class TuiSvgExample1 {
-    readonly timeout$ = timer(0).pipe(mapTo(true));
+    readonly timeout$ = timer(0).pipe(map(() => true));
```
Comment (nsbarsukov) [inline projects/demo/src/modules/components/svg/examples/1/index.ts:19]:
```suggestion
    readonly timeout$ = timer(0).pipe(map(ALWAYS_TRUE_HANDLER));
```

File: projects/kit/components/carousel/carousel.directive.ts [hunk 7]
```diff
@@ -11,10 +11,18 @@ export class TuiCarouselDirective extends Observable<unknown> {
     private readonly duration$ = new BehaviorSubject(0);
 
     private readonly running$ = merge(
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'mouseenter').pipe(mapTo(false)),
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchstart').pipe(mapTo(false)),
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchend').pipe(mapTo(true)),
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'mouseleave').pipe(mapTo(true)),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'mouseenter').pipe(
+            map(() => false),
```
Comment (nsbarsukov) [inline projects/kit/components/carousel/carousel.directive.ts:15]:
```suggestion
            map(ALWAYS_FALSE_HANDLER),
```

File: projects/kit/components/carousel/carousel.directive.ts [hunk 8]
```diff
@@ -11,10 +11,18 @@ export class TuiCarouselDirective extends Observable<unknown> {
...
+        ),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchstart').pipe(
+            map(() => false),
```
Comment (nsbarsukov) [inline projects/kit/components/carousel/carousel.directive.ts:18]:
```suggestion
            map(ALWAYS_FALSE_HANDLER),
```

File: projects/kit/components/carousel/carousel.directive.ts [hunk 9]
```diff
@@ -11,10 +11,18 @@ export class TuiCarouselDirective extends Observable<unknown> {
...
+        ),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchend').pipe(
+            map(() => true),
```
Comment (nsbarsukov) [inline projects/kit/components/carousel/carousel.directive.ts:21]:
```suggestion
            map(ALWAYS_TRUE_HANDLER),
```

File: projects/kit/components/carousel/carousel.directive.ts [hunk 10]
```diff
@@ -11,10 +11,18 @@ export class TuiCarouselDirective extends Observable<unknown> {
...
+        ),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'mouseleave').pipe(
+            map(() => true),
```
Comment (nsbarsukov) [inline projects/kit/components/carousel/carousel.directive.ts:24]:
```suggestion
            map(ALWAYS_TRUE_HANDLER),
```

File: projects/kit/components/slider/helpers/slider-readonly.directive.ts [hunk 11]
```diff
@@ -53,9 +53,9 @@ export class TuiSliderReadonlyDirective {
         const shouldPreventMove$ = merge(
             touchStart$.pipe(
                 tap(e => this.preventEvent(e)),
-                mapTo(true),
+                map(() => true),
```
Comment (nsbarsukov) [inline projects/kit/components/slider/helpers/slider-readonly.directive.ts:56]:
```suggestion
                map(ALWAYS_TRUE_HANDLER),
```

File: projects/kit/components/slider/helpers/slider-readonly.directive.ts [hunk 12]
```diff
@@ -53,9 +53,9 @@ export class TuiSliderReadonlyDirective {
...
             ),
-            touchEnd$.pipe(mapTo(false)),
+            touchEnd$.pipe(map(() => false)),
```
Comment (nsbarsukov) [inline projects/kit/components/slider/helpers/slider-readonly.directive.ts:58]:
```suggestion
            touchEnd$.pipe(map(ALWAYS_FALSE_HANDLER)),
```

File: projects/addon-doc/src/components/navigation/navigation.component.ts [hunk 13]
```diff
@@ -54,6 +54,8 @@ export class TuiDocNavigationComponent {
     );
 
     readonly mode$: Observable<TuiBrightness> = this.mode.change$.pipe(
+        // startWith(null) is not deprecated despite the notice,
+        // but IDE detects a wrong function signature, which is deprecated
```
Comment (nsbarsukov) [inline projects/addon-doc/src/components/navigation/navigation.component.ts:58]: I am not sure that we need to add comment here.

This bug will be fixed in future, but this comment become some black-box unclear comment)

File: projects/addon-doc/src/internal/header/header.component.ts [hunk 14]
```diff
@@ -18,7 +18,7 @@ export class TuiDocHeaderComponent {
     private readonly stream$ = new Subject<boolean>();
 
     readonly open$ = merge(
-        this.router.events.pipe(mapTo(false)),
+        this.router.events.pipe(map(() => false)),
```
Comment (nsbarsukov) [inline projects/addon-doc/src/internal/header/header.component.ts:21]:
```suggestion
        this.router.events.pipe(map(ALWAYS_FALSE_HANDLER)),
```

https://github.com/Tinkoff/taiga-ui/blob/4b799b5f5fb98eeda678b5aa7af82f9901544667/projects/cdk/constants/always-false-handler.ts#L4

File: projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts [hunk 15]
```diff
@@ -18,9 +18,9 @@ export const TUI_SHEET_PROVIDERS: Provider[] = [
         useFactory: ({nativeElement}: ElementRef<HTMLElement>): Observable<boolean> => {
             return merge(
                 tuiTypedFromEvent(nativeElement, `touchstart`, {passive: true}).pipe(
-                    mapTo(true),
+                    map(() => true),
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts:21]:
```suggestion
                    map(ALWAYS_TRUE_HANDLER),
```

https://github.com/Tinkoff/taiga-ui/blob/4b799b5f5fb98eeda678b5aa7af82f9901544667/projects/cdk/constants/always-true-handler.ts#L4

File: projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts [hunk 16]
```diff
@@ -18,9 +18,9 @@ export const TUI_SHEET_PROVIDERS: Provider[] = [
...
                 ),
-                tuiTypedFromEvent(nativeElement, `touchend`).pipe(mapTo(false)),
+                tuiTypedFromEvent(nativeElement, `touchend`).pipe(map(() => false)),
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts:23]:
```suggestion
                tuiTypedFromEvent(nativeElement, `touchend`).pipe(map(ALWAYS_FALSE_HANDLER)),
```

File: projects/addon-mobile/components/sheet/ios.hacks.ts [hunk 17]
```diff
@@ -63,11 +61,16 @@ export function processDragged(
     const race$ = race(scroll$, timer(100)).pipe(
         debounceTime(200),
         take(1),
-        mapTo(false),
+        map(() => false),
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/sheet/ios.hacks.ts:64]:
```suggestion
        map(ALWAYS_FALSE_HANDLER),
```

File: projects/cdk/observables/drag-and-drop-from.ts [hunk 18]
```diff
@@ -35,6 +35,8 @@ export function tuiDragAndDropFrom(element: Element): Observable<TuiDragState> {
             tuiMouseDragFinishFrom(ownerDocument).pipe(
                 take(1),
                 map(event => new TuiDragState(TuiDragStage.End, event)),
+                // endWith(null) is not deprecated despite the notice,
+                // but IDE detects a wrong function signature, which is deprecated
```
Comment (nsbarsukov) [inline projects/cdk/observables/drag-and-drop-from.ts:39]: I am not sure that we need to add comment here.

This bug will be fixed in future, but this comment become some black-box unclear comment)

File: projects/cdk/observables/focus-visible-observable.ts [hunk 19]
```diff
@@ -66,7 +66,13 @@ export function tuiFocusVisibleObservable(element: Element): Observable<boolean>
             filter(isUserFocus => !isUserFocus),
         ),
     ).pipe(
-        switchMapTo(elementBlur$.pipe(mapTo(false), take(1), startWith(true))),
+        switchMap(() =>
+            elementBlur$.pipe(
+                map(() => false),
```
Comment (nsbarsukov) [inline projects/cdk/observables/focus-visible-observable.ts:71]:
```suggestion
                map(ALWAYS_FALSE_HANDLER),
```

File: projects/cdk/observables/is-alive.ts [hunk 20]
```diff
@@ -1,12 +1,17 @@
...
 
 /**
  * Operator to set lifespan after which current value is considered obsolete
  */
 export function tuiIsAlive(lifespan: number = 0): OperatorFunction<unknown, boolean> {
     return pipe(
-        switchMapTo(timer(lifespan).pipe(mapTo(false), startWith(true))),
+        switchMap(() =>
+            timer(lifespan).pipe(
+                map(() => false),
```
Comment (nsbarsukov) [inline projects/cdk/observables/is-alive.ts:11]:
```suggestion
                map(ALWAYS_FALSE_HANDLER),
```

File: scripts/eslint/typescript.js [hunk 21]
```diff
@@ -14,6 +14,37 @@ module.exports = {
             rules: {
                 '@typescript-eslint/quotes': ['error', 'backtick'],
                 '@typescript-eslint/prefer-nullish-coalescing': 'off',
+                'no-restricted-syntax': [
+                    'error',
+                    {
+                        selector: "CallExpression[callee.name='mapTo']",
+                        message:
+                            'Use `map(() => value)` instead of `mapTo(value)`, the operator is deprecated',
```
Comment (nsbarsukov) [inline scripts/eslint/typescript.js:22]: 🚀

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/cdk/tokens/active-element.ts (hunk 1 above)
Comment (splincode) [reply projects/cdk/tokens/active-element.ts]: fixed

File: projects/core/components/alert/alert.component.ts (hunk 2 above)
Comment (splincode) [reply projects/core/components/alert/alert.component.ts]: fixed

File: projects/core/components/link/link.component.ts (hunk 3 above)
Comment (splincode) [reply projects/core/components/link/link.component.ts]: fixed

File: projects/core/components/link/link.component.ts (hunk 4 above)
Comment (splincode) [reply projects/core/components/link/link.component.ts]: fixed

File: projects/demo/src/modules/components/slider/examples/6/index.ts (hunk 5 above)
Comment (splincode) [reply projects/demo/src/modules/components/slider/examples/6/index.ts]: fixed

File: projects/demo/src/modules/components/svg/examples/1/index.ts (hunk 6 above)
Comment (splincode) [reply projects/demo/src/modules/components/svg/examples/1/index.ts]: fixed

File: projects/kit/components/carousel/carousel.directive.ts (hunk 7 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.directive.ts]: fixed

File: projects/kit/components/carousel/carousel.directive.ts (hunk 8 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.directive.ts]: fixed

File: projects/kit/components/carousel/carousel.directive.ts (hunk 9 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.directive.ts]: fixed

File: projects/kit/components/carousel/carousel.directive.ts (hunk 10 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.directive.ts]: fixed

File: projects/kit/components/slider/helpers/slider-readonly.directive.ts (hunk 11 above)
Comment (splincode) [reply projects/kit/components/slider/helpers/slider-readonly.directive.ts]: fixed

File: projects/kit/components/slider/helpers/slider-readonly.directive.ts (hunk 12 above)
Comment (splincode) [reply projects/kit/components/slider/helpers/slider-readonly.directive.ts]: fixed

File: projects/addon-doc/src/components/navigation/navigation.component.ts (hunk 13 above)
Comment (splincode) [reply projects/addon-doc/src/components/navigation/navigation.component.ts]: fixed

File: projects/addon-doc/src/internal/header/header.component.ts (hunk 14 above)
Comment (splincode) [reply projects/addon-doc/src/internal/header/header.component.ts]: fixed

File: projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts (hunk 15 above)
Comment (splincode) [reply projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts]: fixed

File: projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts (hunk 16 above)
Comment (splincode) [reply projects/addon-mobile/components/sheet/components/sheet/sheet.providers.ts]: fixed

File: projects/addon-mobile/components/sheet/ios.hacks.ts (hunk 17 above)
Comment (splincode) [reply projects/addon-mobile/components/sheet/ios.hacks.ts]: fixed

File: projects/cdk/observables/drag-and-drop-from.ts (hunk 18 above)
Comment (splincode) [reply projects/cdk/observables/drag-and-drop-from.ts]: fixed

File: projects/cdk/observables/focus-visible-observable.ts (hunk 19 above)
Comment (splincode) [reply projects/cdk/observables/focus-visible-observable.ts]: fixed

File: projects/cdk/observables/is-alive.ts (hunk 20 above)
Comment (splincode) [reply projects/cdk/observables/is-alive.ts]: fixed

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/kit/components/carousel/carousel.directive.ts [hunk 22]
```diff
@@ -11,10 +15,18 @@ export class TuiCarouselDirective extends Observable<unknown> {
...
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchend').pipe(mapTo(true)),
-        tuiTypedFromEvent(this.elementRef.nativeElement, 'mouseleave').pipe(mapTo(true)),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'mouseenter').pipe(
+            map(ALWAYS_FALSE_HANDLER),
+        ),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchstart').pipe(
+            map(ALWAYS_FALSE_HANDLER),
+        ),
+        tuiTypedFromEvent(this.elementRef.nativeElement, 'touchend').pipe(
+            map(ALWAYS_FALSE_HANDLER),
```
Comment (nsbarsukov) [inline projects/kit/components/carousel/carousel.directive.ts:25]: mistake

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/kit/components/carousel/carousel.directive.ts (hunk 22 above)
Comment (splincode) [reply projects/kit/components/carousel/carousel.directive.ts]: fixed

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**